docs/csharp/programming-guide/inside-a-program/identifier-names.md

```diff
@@ -0,0 +1,44 @@
+---
+title: "Identifier names"
+description: "Learn the rules for valid identifier names in the C# programming language."
+ms.date: 08/21/2018
+---
+# Identifier names
+
+An **identifier** is the name you assign to a type (class, interface, struct, delegate, or event), member, variable, or namespace. Valid identifiers must follow these rules:
+
+- Identifiers must start with a letter, or `_`.
+- Identifiers may contain Unicode letter characters, decimal digit characters, Unicode connecting characters, Unicode combining characters, or Unicode formatting characters. For more information on Unicode categories, see the [Unicode Category Database](https://www.unicode.org/reports/tr44/).
+- Identifiers should not contain two consecutive `_` characters. Those are reserved for compiler generated identifiers.
```

Comment:
@BillWagner the last point sounds more like a convention (indeed, one can declare `a__b` variable). Or you want present it as a rule?

Reply:
I'd move it to an "In addition..." paragraph in the naming conventions section, since it is a convention and not a requirement.